File: src/main/java/com/thatameliah/RFIDRegister/Handlers/FileHandler.java
package com.thatameliah.RFIDRegister.Handlers;

import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class FileHandler {
    private static final Path SAVES_DIRECTORY = Paths.get("saves");

    // Gets the Path to the specified file
    private Path GetPathToFile(String filename, String extension) {
        return SAVES_DIRECTORY.resolve(filename + extension);
    }

    // Ensure the "saves" directory exists
    private void EnsureDirectory() throws IOException {
        if (!Files.exists(SAVES_DIRECTORY)) {
            Files.createDirectories(SAVES_DIRECTORY);
        }
    }

    // Write a string to a file
    public void WriteToFile(String content, String filename, String extension) {
        Path path = GetPathToFile(filename, extension);
        try {
            EnsureDirectory();
            try (BufferedWriter writer = new BufferedWriter(new FileWriter(path.toFile()))) {
                writer.write(content);
            }
            System.out.println("Successfully wrote to " + path);
        } catch (IOException e) {
            System.err.println("Error writing to file " + path + ": " + e.getMessage());
        }
    }

    // Read file content as string
    public String ReadFile(String filename, String extension) {
        Path path = GetPathToFile(filename, extension);
        if (!Files.exists(path)) {
            System.err.println("File not found: " + path);
            return null;
        }
        try {
            return Files.readString(path);
        } catch (IOException e) {
            System.err.println("Error reading file " + path + ": " + e.getMessage());
            return null;
        }
    }

    // Delete file if exists
    public void DeleteFile(String filename, String extension) {
        Path path = GetPathToFile(filename, extension);
        try {
            if (Files.deleteIfExists(path)) {
                System.out.println("Deleted file " + path);
            }
        } catch (IOException e) {
            System.err.println("Failed to delete file " + path + ": " + e.getMessage());
        }
    }
}

File: src/main/java/com/thatameliah/SmartCardRegister/Utils/FileHandler.scala
package com.thatameliah.SmartCardRegister.Utils

import org.jetbrains.annotations.{NotNull, Nullable}

import java.io.{File, IOException}
import java.nio.file.{Files, Path}

object FileHandler {
  /**
   * Writes a string to a file
   * @param content The String to be written
   * @param file The file to write to
   * @throws NullPointerException If file or content are null
   */
  def WriteToFile(@NotNull file: File, @NotNull content: String): Unit = {
    if (file == null || content == null) throw new NullPointerException
    val path = file.toPath
    
    try {
      Files.createDirectories(path.getParent)
      Files.writeString(path, content)
    }
    catch {
      case err: IOException =>
        System.err.println("Error writing to file " + path.toString + ": " + err.getMessage)
    }
  }

  /**
   * Reads the contents of a file
   * @param file The file to read from
   * @return The string contents of the file
   */
  @NotNull def ReadFile(@NotNull file: File): String = {
    val path: Path = file.toPath

    if (!Files.exists(path)) { return new String }
    try { Files.readString(path) }
    catch { case _: IOException => new String }
  }
}